tests/Doctrine/Annotation/TokenTest.php

```diff
@@ -0,0 +1,152 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Doctrine\Annotation;
+
+use Doctrine\Common\Annotations\DocLexer;
+use PhpCsFixer\Doctrine\Annotation\Token;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Doctrine\Annotation\Token
+ */
+final class TokenTest extends TestCase
+{
+    public function testDefaults()
+    {
+        $token = new Token();
+
+        $this->assertSame(DocLexer::T_NONE, $token->getType());
+        $this->assertSame('', $token->getContent());
+    }
+
+    public function testConstructorSetsValues()
+    {
+        $type = 42;
+        $content = 'questionable';
+
+        $token = new Token(
+            $type,
+            $content
+        );
+
+        $this->assertSame($type, $token->getType());
+        $this->assertSame($content, $token->getContent());
+    }
+
+    public function testCanModifyType()
+    {
+        $type = 42;
+
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertSame($type, $token->getType());
+    }
+
+    /**
+     * @dataProvider providerIsType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsTrue($type, $types)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertTrue($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsType()
+    {
+        return array(
+            'same-value' => array(
+                42,
+                42,
+            ),
+            'array-with-value' => array(
+                42,
+                array(
+                    42,
+                    9001,
+                ),
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerIsNotType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsFalse($types, $type)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertFalse($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsNotType()
+    {
+        return array(
+            'different-value' => array(
+                42,
+                9001,
+            ),
+            'array-without-value' => array(
+                42,
+                array(
+                    9001,
+                ),
+            ),
+        );
+    }
+
+    public function testCanModifyContent()
+    {
+        $content = 'questionable';
+
+        $token = new Token();
+
+        $token->setContent($content);
+
+        $this->assertSame($content, $token->getContent());
+    }
+
+    public function testCanClearContent()
+    {
+        $content = 'questionable';
+
+        $token = new Token();
+
+        $token->setContent($content);
```

Comment:
question on concept level @keradus and @julienfalque ; are we going to deprecate this for ` PhpCsFixer\Doctrine\Annotation\Token;` as well?